plugins/scaffolder-backend/src/ScaffolderPlugin.ts

```diff
@@ -23,120 +23,104 @@ import { ScmIntegrations } from '@backstage/integration';
 import { catalogServiceRef } from '@backstage/plugin-catalog-node/alpha';
 import {
   scaffolderActionsExtensionPoint,
-  ScaffolderActionsExtensionPoint,
+  scaffolderTaskBrokerExtensionPoint,
+  scaffolderTemplatingExtensionPoint,
+  TaskBroker,
   TemplateAction,
-} from '@backstage/plugin-scaffolder-node';
-import {
   TemplateFilter,
   TemplateGlobal,
-  TaskBroker,
-} from '@backstage/plugin-scaffolder-backend';
+} from '@backstage/plugin-scaffolder-node';
 import { createBuiltinActions } from './scaffolder';
 import { createRouter } from './service/router';
 
 /**
- * Catalog plugin options
+ * Scaffolder plugin
  *
  * @alpha
  */
-export type ScaffolderPluginOptions = {
-  actions?: TemplateAction<any, any>[];
-  taskWorkers?: number;
-  taskBroker?: TaskBroker;
-  additionalTemplateFilters?: Record<string, TemplateFilter>;
-  additionalTemplateGlobals?: Record<string, TemplateGlobal>;
-};
+export const scaffolderPlugin = createBackendPlugin({
+  pluginId: 'scaffolder',
+  register(env) {
+    const addedActions = new Array<TemplateAction<any, any>>();
+    env.registerExtensionPoint(scaffolderActionsExtensionPoint, {
+      addActions(...newActions: TemplateAction<any>[]) {
+        addedActions.push(...newActions);
+      },
+    });
 
-class ScaffolderActionsExtensionPointImpl
-  implements ScaffolderActionsExtensionPoint
-{
-  #actions = new Array<TemplateAction<any, any>>();
+    let taskBroker: TaskBroker | undefined;
+    env.registerExtensionPoint(scaffolderTaskBrokerExtensionPoint, {
+      setTaskBroker(newTaskBroker) {
+        if (taskBroker) {
+          throw new Error('Task broker may only be set once');
+        }
+        taskBroker = newTaskBroker;
+      },
+    });
 
-  addActions(...actions: TemplateAction<any>[]): void {
-    this.#actions.push(...actions);
-  }
+    const additionalTemplateFilters: Record<string, TemplateFilter> = {};
+    const additionalTemplateGlobals: Record<string, TemplateGlobal> = {};
+    env.registerExtensionPoint(scaffolderTemplatingExtensionPoint, {
+      addTemplateFilters(newFilters) {
```

Comment:
just noting that we sometimes tend to prefer add-single-thing-at-a-time interfaces, but this is fine

Reply:
Yeah was considering redesigning this a bit but in the end figured this was alright too